src/engine/bufferscalers/rubberbandtask.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <rubberband/RubberBandStretcher.h>
+
+#include <QMutex>
+#include <QRunnable>
+#include <QWaitCondition>
+#include <atomic>
+
+#include "audio/types.h"
+
+using RubberBand::RubberBandStretcher;
+
+class RubberBandTask : public RubberBandStretcher, public QRunnable {
+  public:
+    RubberBandTask(size_t sampleRate,
+            size_t channels,
+            Options options = DefaultOptions);
+
+    /// @brief Submit a new stretching task
+    /// @param stretcher the RubberBand::RubberBandStretcher instance to use.
+    /// Must remain valid till waitReady() as returned
+    /// @param input The samples buffer
+    /// @param samples the samples count
+    /// @param final whether or not this is the final buffer
+    void set(const float* const* input,
+            size_t samples,
+            bool isFinal);
+
+    // Wait for the current task to complete.
+    void waitReady();
+
+    void run();
+
+  private:
+    // Used to schedule the thread
+    QMutex m_waitLock;
+    QWaitCondition m_waitCondition;
```

Comment:
I think we can Remove the QWaitCondition because we have only a two threads that needs to be synched. 
I think a better solution would be to make m_completed a QSemsphore. It's cooky jar model looks probably nicer.

Reply:
I believe we can remove the lock too in that case (`set` and and `wait` are called from the main thread).